components/notifier/src/strategies/reds_that_are_new.py

```diff
@@ -0,0 +1,28 @@
+"""Strategies for notifying users about metrics."""
+
+from typing import List
+
+
+def reds_that_are_new(json) -> List:
+    """Return the number of red metrics per report in the JSON returned by /api/v3/reports."""
+    new_red_metrics = []
+    for report in json["reports"]:
+        new_red_metrics.append({
+            "report_uuid": report["report_uuid"],
+            "report_title": report["title"],
+            "teams_webhook": report.get("teams_webhook", ""),
+            "url": report.get("url", ""),
+            "new_red_metrics": 0})
+        for subject in report["subjects"].values():
+            for metric in subject["metrics"].values():
+                if metric["status"] == "target_not_met" and used_to_be_not_red(metric):
+                    new_red_metrics[-1]["new_red_metrics"] += 1
+
+    return new_red_metrics
+
+
+def used_to_be_not_red(metric):
+    """determine if a metric used to be something other than red"""
+    if "recent_measurements" in metric:
+        return metric["recent_measurements"][0]["count"]["status"] != "target_not_met"
```

Comment:
This isn't correct: you're looking at the first recent measurement, which is the oldest recent measurement. You have to look at the last recent measurement, .i.e. `metric["recent_measurements"][-1]` and check whether it is red **and** has a start and end time that are equal. You'll have to look at your unit tests as well. If they are passing they are incorrect as well...

Reply:
replaced the comparison between the status of the metric and the most recent measurement with a comarison between the start and end of the most recent measurement